src/seedu/addressbook/data/person/Address.java

```diff
@@ -6,22 +6,21 @@
  * Represents a Person's address in the address book.
  * Guarantees: immutable; is valid as declared in {@link #isValidAddress(String)}
  */
-public class Address {
+public class Address extends Contact {
 
     public static final String EXAMPLE = "123, some street";
     public static final String MESSAGE_ADDRESS_CONSTRAINTS = "Person addresses can be in any format";
     public static final String ADDRESS_VALIDATION_REGEX = ".+";
 
     public final String value;
-    private boolean isPrivate;
 
     /**
      * Validates given address.
      *
      * @throws IllegalValueException if given address string is invalid.
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
-        this.isPrivate = isPrivate;
+        super.isPrivate = isPrivate;
```

Comment:
in each of the subclasses, you should not be setting super.isPrivate directly, rather your constructor should have parameters that take in all the common elements such as value and isPrivate.